### PR TITLE
[Filesystem] Fixes #4754 - Use express' attachment method and reliably parse the content-disposition

### LINKED ISSUE #4754: [Filesystem] 500 Error when downloading files with non-ascii characters
### Description
Theia does not allow downloading of files/folders that uses some non-ascii characters on their filenames/folder names.

### Reproduction Steps
- Create a new file and name it `Φtest.txt` (you can copy/paste that and use as the filename)
- Once created download the file, Right-click -> download. It will not do anything, and if you check your browser's console, you will see an error which is a `500 error`

Screenshot:
image

**OS and Theia version:**
Win10 - Theia Master branch

### PR DESCRIPTION
**Pull request by uniibu**
This commit fixes https://github.com/theia-ide/theia/issues/4754 and also conforms to latest rfc specs

Signed-off-by: Uni Sayo <unibtc@gmail.com>

### CODE REVIEW

File: packages/filesystem/src/common/download/file-download-disposition.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+export interface ContentDispositionParams {
+    filename: string;
+    'filename*'?: string;
+    'name'?: string;
+    [propName: string]: string | undefined;
+}
+
+export namespace FileDownloadContentDisposition {
```
Comment (akosyakov) [inline packages/filesystem/src/common/download/file-download-disposition.ts]: It should be a class, see https://github.com/theia-ide/theia/wiki/Coding-Guidelines#di-function-export

**Comment by akosyakov** [comment]
@kittaakos please have a look

**Comment by uniibu** [comment]
@kittaakos The problem with `encodeURIComponent` on `Content-Disposition` is browser compatibility. It works fine on Chrome, but like in Firefox it has to be set as ` filename*="utf-8''${encodeURIComponent(name)}" ` [example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#Examples) and there are also other incompatibilities on other browsers such as safari and ie when simply using encodeURIComponent. Another filename you could test are with chinese characters like `中指南.txt`

**Comment by kittaakos** [comment]
> safari and ie

They are out of the scope.

> Another filename you could test are with chinese characters like `中指南.txt`

OK, I will but later, please bear with me.

**Comment by uniibu** [comment]
messed up the squash. redoing on new PR

**Pull request closed without merging**